File: src/main/java/it/astromark/attendance/dto/AttendanceResponse.java
package it.astromark.attendance.dto;

import io.swagger.v3.oas.annotations.media.Schema;

import java.time.Instant;
import java.util.UUID;

@Schema(description = "Response object containing attendance details for a student")
public record AttendanceResponse(
        UUID id,
        String name,
        String surname,
        Boolean isAbsent,
        Boolean isDelayed,
        Instant delayTime,
        Boolean delayNeedJustification,
        Integer totalAbsence,
        Integer totalDelay
) {
}

File: src/main/typescript/src/entities/AttendanceResponse.ts
import {DateObject} from "react-multi-date-picker";

export type AttendanceResponse = {
    id: string;
    name: string;
    surname: string;
    isAbsent: boolean;
    isDelayed: boolean;
    delayTime: DateObject;
    delayNeedJustification: boolean;
    totalAbsence: number;
    totalDelay: number;
}